api/v1beta1/openstackmachine_types.go

```diff
@@ -33,6 +33,60 @@ const (
 	IPClaimMachineFinalizer = "openstackmachine.infrastructure.cluster.x-k8s.io/ip-claim"
 )
 
+// SchedulerHintValueType is the type that represents allowed values for the Type field.
+// +kubebuilder:validation:Enum=Bool;String;Number
+type SchedulerHintValueType string
+
+// Constants representing the allowed types for SchedulerHintAdditionalValue.
+const (
+	SchedulerHintTypeBool   SchedulerHintValueType = "Bool"
+	SchedulerHintTypeString SchedulerHintValueType = "String"
+	SchedulerHintTypeNumber SchedulerHintValueType = "Number"
+)
+
+// SchedulerHintAdditionalValue represents the value of a scheduler hint property.
+// The value can be of various types: Bool, String, or Number.
+// The Type field indicates the type of the value being used.
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'Bool' ? has(self.bool) : !has(self.bool)",message="bool is required when type is Bool, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'Number' ? has(self.number) : !has(self.number)",message="number is required when type is Number, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'String' ? has(self.string) : !has(self.string)",message="string is required when type is String, and forbidden otherwise"
+// +union.
+type SchedulerHintAdditionalValue struct {
+	// Type represents the type of the value (Bool, String, or Number).
+	// +kubebuilder:validation:Required
+	// +unionDiscriminator
+	Type SchedulerHintValueType `json:"type"`
+
+	// Bool is the boolean value of the scheduler hint, used when Type is "Bool".
+	// This field is required if type is 'Bool', and must not be set otherwise.
+	// +unionMember,optional
+	Bool *bool `json:"bool,omitempty"`
+
+	// Number is the integer value of the scheduler hint, used when Type is "Number".
+	// This field is required if type is 'Number', and must not be set otherwise.
+	// +unionMember,optional
+	Number *int `json:"number,omitempty"`
+
+	// String is the string value of the scheduler hint, used when Type is "String".
+	// This field is required if type is 'String', and must not be set otherwise.
+	// +unionMember,optional
+	String *string `json:"string,omitempty"`
```

Comment:
Is the empty string a valid value? Currently it will pass validation without a minLength of 1

Can you impost a maximal string length to this?